src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,15 +1,144 @@
-import { Component } from 'react';
+import React from 'react';
 
-type Props = {};
-type State = {};
+import './NewMovie.scss';
 
-export class NewMovie extends Component<Props, State> {
-  state: State = {};
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+type State = {
+  title: string,
+  description: string,
+  imgUrl: string,
+  imdbUrl: string,
+  imdbId: string,
+};
+
+type EventType = React.ChangeEvent<HTMLInputElement>
+| React.ChangeEvent<HTMLTextAreaElement>;
+
+export class NewMovie extends React.Component<Props, State> {
+  state = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  };
+
+  inputHandler = (event: EventType) => {
+    const { name, value } = event.target;
+
+    this.setState(prevState => ({
+      ...prevState,
+      [name]: value,
+    }));
+  };
+
+  submitHandler = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const {
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    } = this.state;
+
+    this.props.onAdd({
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    });
```

Comment:
Возможно, я не полностью разобрался в твоей задумке, но в аргумент onAdd мы просто можем передать this.state из компонента NewMovie, поскольку он по факту совпадает с типом Movie.
```suggestion
    this.props.onAdd(this.state);
```